### PR TITLE
Fix sonar bug

### PR DESCRIPTION
**Pull request by jt2594838**

### CODE REVIEW

**Comment by jixuan1989** [comment]
```
[INFO] Running org.apache.iotdb.db.engine.overflow.io.OverflowProcessorTest
INFO  [2019-03-01 09:35:08,331] [main] org.apache.iotdb.db.auth.user.BasicUserManager:77 - Admin initialized 
INFO  [2019-03-01 09:35:08,333] [main] org.apache.iotdb.db.auth.authorizer.BasicAuthorizer:63 - Initialization of Authorizer completes 
WARN  [2019-03-01 09:35:08,335] [main] org.apache.iotdb.db.engine.overflow.io.OverflowResource:130 - Cannot read position info, returning a default value 
java.io.FileNotFoundException: data/data/overflow/test/0/positionFile (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
	at org.apache.iotdb.db.engine.overflow.io.OverflowResource.readPositionInfo(OverflowResource.java:109)
```
We need to care about this failed UT.

File: iotdb/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 1]
```diff
@@ -291,37 +294,37 @@ public void updatePath() {
     preUpdatePath();
 
     // update the paths of subdirectories in the dataDir
-    if (dataDir.length() > 0 && !dataDir.endsWith(File.separator)) {
-      dataDir = dataDir + File.separatorChar;
+    if (getDataDir().length() > 0 && !getDataDir().endsWith(File.separator)) {
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java]: do we need use getDataDir() here?

Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java]: This is automatically refactored. And I think this won't hurt a bit.

**Comment by jixuan1989** [comment]
```
[ERROR] Failures: 
[ERROR]   IoTDBMetadataFetchIT.databaseMetaDataTest:226->showTimeseriesInJson:385 expected:<...        Compressor: [UNCOMPRESSED,
                     args: {max_point_number=3, compressor=SNAPPY]},
                 ...> but was:<...        Compressor: [SNAPPY,
                     args: {max_point_number=3]},
                 ...>
[INFO] 
[ERROR] Tests run: 50, Failures: 1, Errors: 0, Skipped: 1
```
notice the error.

In a merged PR, I changed the output of time series schema, which put compressor out of args. You may need to merge with origin/master to fix this error.

**Pull request merged**